### ISSUE TITLE
Create dependencies between issues

### ISSUE THREAD

**Issue by aaronwinter** [waffleio/waffle.io#1996]
It would be great to enable users to link different issues together. For example if issue A needs issue B, I would like to see that dependency chain.

**Comment by tylerwgoza** [comment]
+1

**Comment by ehausen** [comment]
Github already has reference functionality.

I think waffle should support the existing functionality fully and not complicate things too much. If waffle starts implement its own system then what is the point to even connect to github issues?

For more complicated issue management there is already very robust products, like Redmine or Jira.

**Comment by aaronwinter** [comment]
It would be nice to integrate Github references in Waffle's UI + dependency
links would let me "hide" issues that have dependencies still unfulfilled.

I am not even using Waffle to manage a midsize project but my board is
already messy, I love this product and think there is a lot of potential in
it but this feature is crucial IMHO.

**Comment by ehausen** [comment]
I think the Github philosophy here would be to use milestones. Just divide the tasks to different milestones, maybe with deadline and then filter by milestone.

**Comment by ehausen** [comment]
But I see your point. I just think the github issues should still be usable also when using waffle

**Comment by ehausen** [comment]
overv.io has a functionality to add time estimate. It just creates a comment: Time estimate: XX and the application reads it.

Maybe there could be something similar:
**Parent issue**
* Parent issue field which would output a comment: "Parent issue: #12345" then it would be fully readable on Github

**Comment by pnpritchard** [comment]
+1

**Comment by TomasMikula** [comment]
+1

**Comment by supertylerc** [comment]
:+1:

**Comment by codewaters** [comment]
+1

**Comment by WanderingMatt** [comment]
+1

Showing any issue dependencies attached to the card would be super helpful. Would love to see Waffle adopt the "Needs [issue link]" comment convention for this.

**Comment by sfarmar** [comment]
+1

**Comment by nhenin** [comment]
+1

**Comment by Ryanb58** [comment]
+1

This would be a good investment as it would give you the ability, later on, to add a milestone based gantt chart with dependencies already enabled. :)

**Comment by dvdstelt** [comment]
How come this is planned, but https://github.com/waffleio/waffle.io/issues/803 is much older and seems somewhat like a duplicate? Although #803 is more concrete. And more needed by our team! :-)

**Comment by dvdstelt** [comment]
Would be nice to see something like PRs do now. If issue #1 is dependent on #2 it would show like a PR. And when you close #1 automatically #2 would become the new issue on top. Based on labels it would automatically end up in the correct lane.

**Comment by homeyer** [comment]
@dvdstelt thanks! #803 also moved to planned. We'll visualize dependencies so it's easy to find things that are waiting or blocked by work. After that we'll investigate how to auto-move things appropriately!

**Comment by kellihsf** [comment]
Hi all, 👋

Waffle now supports dependencies!! 🎉 

Read more about dependencies here: [Help Center](https://help.waffle.io/dependencies) 

Email us at support@waffle.io with any questions or feedback! 

![dep-gif-1-1_00000.gif](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/56241220b728ed1d0009af87/125516c66e82c728ace21e0d46ec918a6e858aafc2b8f2129b0da0c47916653f6b14b1247a80e03cfd1839545d0a4ea8400b0911fcfd8c79aee76f738f0a5ee9d76f5f66da2f53fde6d640e258f075c95d8fa4bd8d46c1ed3f23e28c8e4b361b001988ed10.gif)

**Comment by mightyiam** [comment]
![giphy](https://user-images.githubusercontent.com/635591/40188334-5ed47b1a-5a02-11e8-83c4-c898703c6021.gif)

**Comment by aaronwinter** [comment]
Thanks for making this happen!
